### DISCUSSION TITLE
issues with converting format and sending to stdout (specifically mp4)

### DISCUSSION THREAD

**Discussion by torcado194** [ImageMagick/ImageMagick#1949] [General] [answered]
Hey there,
I'm facing an identical issue on 3 different machines: windows 10, debian 10.0, and ubuntu 18.04. I have encountered the issue on imagemagick versions 6.9.2-0, 7.0.7-39, 7.0.9-24, and 7.0.10-9

The goal is to create an mp4 video and send the data to stdout

for the simple command:
```
convert -delay 1x50 rose: -duplicate 49 mp4:-
```

I either get an error like:
```
convert: delegate failed `"ffmpeg.exe" -nostdin -v -1 -i "%M%%d.jpg" "%u.%m"' @ error/delegate.c/InvokeDelegate/1897.
```
or it creates a file named `-` with the correct data, and changing the extension to .mp4 and opening it displays just as it should.

replacing `mp4:` with other formats like `miff:` produce the desired result:
```
convert -delay 1x50 rose: -duplicate 49 miff:-
```

additionally, simply piping an existing mp4 file to stdout:
```
convert test.mp4 mp4:-
```
causes the same issues, either creating a file named `-` or showing a `delegate failed` error

how do I send an mp4 file to stdout?
thanks

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by torcado194** [comment]
great, thank you very much!
